config.py

```diff
@@ -149,6 +150,37 @@ def _setup_logging() -> None:
         logger.addHandler(console_logging_handler)
         logger.propagate = False
 
+    @staticmethod
+    def _setup_discord_log_level() -> None:
+        raw_discord_log_level: str = os.getenv("DISCORD_LOG_LEVEL", "").upper().strip()
+
+        if not raw_discord_log_level:
+            logger.debug("DISCORD_LOG_LEVEL is not set, skipping Discord logging setup.")
+            return
+
+        if raw_discord_log_level not in LOG_LEVEL_CHOICES:
+            INVALID_LOG_LEVEL_MESSAGE: Final[str] = (
+                "DISCORD_LOG_LEVEL must be one of "
+                f"{
+                    ','.join(
+                        f'{log_level_choice!r}' for log_level_choice in LOG_LEVEL_CHOICES[:-1]
+                    )
+                }"
+            )
+            raise ImproperlyConfiguredError(INVALID_LOG_LEVEL_MESSAGE)
+
+        discord_logger.setLevel(getattr(logging, raw_discord_log_level))
+
+        discord_log_handler = logging.FileHandler(
+            filename="discord.log", encoding="utf-8", mode="w"
```

Comment:
```suggestion
            filename="discord.log", encoding="utf-8", mode="a"
```

I think mode `a` will append that might fix the issue that the log file wasn't getting written to? I haven't read the docs and this is just an assumption though.